### PR TITLE
Add plistlib

### PR DESCRIPTION
**Pull request by tharvik**
https://docs.python.org/3/library/plistlib.html

### CODE REVIEW

**Comment by tharvik** [comment]
Fail because of [pytype#34](https://github.com/google/pytype/issues/34).

**Comment by matthiaskramm** [comment]
Oops, we'll take a look at https://github.com/google/pytype/issues/34.

In the meantime, you can work around the issue by doing
```
mm = MutableMapping[str, Any]
_D = TypeVar('_D', mm)
```

**Comment by tharvik** [comment]
@matthiaskramm that indeed remove [pytype#34](https://github.com/google/pytype/issues/34), but triggers [pytype#33](https://github.com/google/pytype/issues/33).

**Comment by matthiaskramm** [comment]
In that case, please just add the affected files to the pytype blacklist. We'll look at this issue soon.

File: stdlib/2and3/plistlib.pyi [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+             use_builtin_types: bool, dict_type: Type[_D] =...) -> _D: ...
+    def loads(data: bytes, *, fmt: Optional[PlistFormat] = ...,
+              use_builtin_types: bool = ..., dict_type: Type[_D] = ...) -> _D: ...
+    def dump(value: Mapping[str, Any], fp: IO[bytes], *,
+             fmt: PlistFormat =..., sort_keys: bool = ...,
+             skipkeys: bool = ...) -> None: ...
+    def dumps(value: Mapping[str, Any], *, fmt: PlistFormat = ...,
+              skipkeys: bool = ..., sort_keys: bool = ...) -> bytes: ...
+
+def readPlist(pathOrFile: Union[str, IO[bytes]]) -> DictT[str, Any]: ...
```
Comment (gvanrossum) [inline stdlib/2and3/plistlib.pyi]: In PY2 this should also accept unicode.

File: stdlib/2and3/plistlib.pyi [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+def writePlist(value: Mapping[str, Any], pathOrFile: Union[str, IO[bytes]]) -> None: ...
+def readPlistFromBytes(data: bytes) -> DictT[str, Any]: ...
```
Comment (gvanrossum) [inline stdlib/2and3/plistlib.pyi]: In PY2 there's also readPlistFromString and writePlistToString.

**Comment by tharvik** [comment]
@matthiaskramm added.
@gvanrossum done.
And CI is happy now.

**Comment by gvanrossum** [comment]
OK, please wait for my manual CI. :-)

**Comment by gvanrossum** [comment]
Hm, you're lucky, it actually did pass. But I hadn't reported back yet...

**Comment by matthiaskramm** [comment]
Oh oops. For some reason, I thought I saw the new status. (The status was displayed *after* your message, not before, making it look as if the latter was more recent). My apologies.

In other news, I think you're the only maintainer who has the magical "rerun travis" button. I always have to close and reopen the PRs.

**Comment by gvanrossum** [comment]
Sorry, I said "manual". I have to run mypy against our internal repos
before vetting a PR. (I haven't been doing this consistently, and I've
suffered badly because of it, see my recent changes to contextlib and
threading.)

Regarding the magical travis re-run button, that may be because I happen to
be the owner of the Travis setup? I find Travis setup confusing, even
though I love what the tool does for us. I don't see a way to add other
people to the setup.

**Comment by matthiaskramm** [comment]
Oh! That's what you meant by "manual"! Sorry. I'll wait for you next time.

**Comment by tharvik** [comment]
@gvanrossum You mention these internal repos numerous times, maybe we can add theses directly to typeshed testsuite?

**Comment by gvanrossum** [comment]
Sadly not, it's proprietary code. I presume Google also occasionally runs pytype against their internal repo to see how it's doing? Maybe we're farther along, in the sense that we do actually require mypy-cleanliness before commits to these internal repos are allowed to go in. That also makes me occasionally jittery when some big typeshed change goes in -- running those can reveal (a) bad annotations, (b) bugs in the stubs, (c) bugs in our internal code (rare, since it's also tested in other ways), or (d) ambiguity regarding what the stubs should do about undocumented APIs/behaviors. (Latest example: contextlib.nested() returns a list of values, but the docs don't say.)

**Comment by gvanrossum** [comment]
Some more failure modes: mypy bugs or missing features, or ambiguities in PEP 484.

**Comment by tharvik** [comment]
It's what I was thinking (there might be some CI-magic that can be done, without showing build logs, building on PR hooks, and stuff but I guess that's not worth the trouble), too bad then.
Precisely documenting the stubs (as I'm trying to do) can lead to much (d), which IMO is a good thing, if it undocumented/badly documented that mean that the code should not really depend on it, so approching (c) in a way. (a) and (b) are usually mistakes during stub creation, but that's a normal part of the process when going from a stub having `Any` everywhere to a typed one (forgot some method from py2 to py3, some names changes, weird subclassing, ...); it's what these PR are for, and applying them and see where it breaks and fix as needed.

**Comment by matthiaskramm** [comment]
Yeah, we have the same problem. Any typeshed update has the potential to break us.

Maybe we should do what you do, and deal with changes PR-by-PR, instead of in bulk whenever we update. Updating typeshed is already a tricky process, and I suspect it'll get harder over time, the more our code-base depends on type-checking with pytype.

**Pull request merged**